integrations/watsonx/src/haystack_integrations/components/embedders/watsonx/document_embedder.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+from ibm_watsonx_ai import Credentials
+from ibm_watsonx_ai.foundation_models import Embeddings
+
+
+@component
+class WatsonXDocumentEmbedder:
+    """
+    Computes document embeddings using IBM watsonx.ai models.
+
+    ### Usage example
+
+    ```python
+    from haystack import Document
+    from haystack.components.embedders import WatsonXDocumentEmbedder
+
+    documents = [
+        Document(content='I love pizza!'),
+        Document(content='Pasta is great too'),
+    ]
+
+    document_embedder = WatsonXDocumentEmbedder(
+        model='ibm/slate-30m-english-rtrvr',
+        api_key=Secret.from_env_var('WATSONX_API_KEY'),
+        url='https://us-south.ml.cloud.ibm.com',
+        project_id='your-project-id',
+    )
+
+    result = document_embedder.run(documents=documents)
+    print(result['documents'][0].embedding)
+
+    # [0.017020374536514282, -0.023255806416273117, ...]
+    ```
+    """
+
+    def __init__(
+        self,
+        model: str = 'ibm/slate-30m-english-rtrvr',
+        api_key: Secret | None = None,
+        url: str = 'https://us-south.ml.cloud.ibm.com',
+        project_id: str | None = None,
+        space_id: str | None = None,
+        truncate_input_tokens: int | None = None,
+        prefix: str = '',
+        suffix: str = '',
+        batch_size: int = 1000,
```

Comment:
Could we also add the options
```python
        meta_fields_to_embed: Optional[List[str]] = None,
        embedding_separator: str = "\n",
```
like we have in our other Document embedders like `OpenAIDocumentEmebdder` (see [here](https://github.com/deepset-ai/haystack/blob/556dcc9e4628abc651e4266d4bc7af5b1ff36f83/haystack/components/embedders/openai_document_embedder.py#L53-L54))

We find being able to prepend metadata to the document content can help with retrieval tasks.